### PR TITLE
Feature/add before each and improve checkboxes

### PR DESCRIPTION
**Pull request by fernandorobles97**
What I did:
- Implemented beforeEach in testing suites.
- Added a div element for every label and input element that makes up a checkbox.
- Added styling rules to make it more clear which checkbox is being clicked.
- Added hidden class to specific buttons when a user is in the section that matches that button.

Did this break anything?
- [x] No
- [ ]  Yes

Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:
- [x]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next?
Address more UI issues.

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -45,42 +45,78 @@ <h1 class = "title">What's Cookin?</h1>
...
-      <input type="checkbox" class="checkbox" id="snack" name="checkbox">
-      <label for="snack">Snack</label>
-      <input type="checkbox" class="checkbox" id="spread" name="checkbox">
-      <label for="spread">Spread</label>
-      <input type="checkbox" class="checkbox" id="starter" name="checkbox">
-      <label for="starter">Starter</label>
+      <div class="filter-option">
+        <input type="checkbox" class="checkbox" id="antipasto" name="checkbox">
+        <label for="antipasto">Antipasto</label>
+      </div>
```
Comment (Denzel-Turambi) [inline dist/index.html:51]: Nice job wrapping the checkbox input and label in a div.

File: test/RecipeUtils-test.js [hunk 2]
```diff
@@ -6,20 +6,28 @@ import { recipesToCook, saveRecipe } from '../src/userUtils';
...
 
 describe('recipesFromTag', () => {
+  let save, save2, expected, names;
+
+  beforeEach(function() {
+    save = saveRecipe(recipeData, 'Loaded Chocolate Chip Pudding Cookie Cups');
+    save2 = saveRecipe(recipeData, 'Maple Dijon Apple Cider Grilled Pork Chops');
+    expected = ['Thai Chicken Tenders with Broiled Pineapple Slaw', 'Maple Dijon Apple Cider Grilled Pork Chops'];
+    names = ['Maple Dijon Apple Cider Grilled Pork Chops', 'Thai Chicken Tenders with Broiled Pineapple Slaw'];
+  })
```
Comment (Denzel-Turambi) [inline test/RecipeUtils-test.js:16]: Nice test refactoring, and use for beforeEach()

**Review by Denzel-Turambi — APPROVED** [review]

File: dist/index.html (hunk 1 above)
Comment (andreasorensen) [reply dist/index.html:51]: This looks really good!

File: src/domUpdates.js [hunk 3]
```diff
@@ -26,13 +26,13 @@ const searchButton = document.querySelector('.search-button')
...
-  addHiddenClass([saveRecipeButton, savedRecipeDisplay, savedSearchInput])
-  removeHiddenClass([searchInput])
+  addHiddenClass([saveRecipeButton, savedRecipeDisplay, savedSearchInput, allRecipesButton])
+  removeHiddenClass([searchInput, savedRecipesButton])
 });
 
 savedRecipesButton.addEventListener('click', () => {
-  addHiddenClass([allRecipeDisplay, singleRecipeDisplay, saveRecipeButton, saveRecipeButton, frontRecipeDisplay, searchInput]);
-  removeHiddenClass([savedRecipeDisplay, savedSearchInput, allFilterDisplay]);
+  addHiddenClass([allRecipeDisplay, singleRecipeDisplay, saveRecipeButton, savedRecipesButton, frontRecipeDisplay, searchInput]);
```
Comment (andreasorensen) [inline src/domUpdates.js:34]: Awesome job hiding those buttons when they're not needed!

File: src/styles.css [hunk 4]
```diff
@@ -149,6 +150,16 @@ aside {
   width: 80vw;
 }
 
+.filter-option {
+  height: 5vh;
+  width: 8vw;
+  border-bottom: 3px solid rgb(214, 214, 214);
+  display: flex;
```
Comment (andreasorensen) [inline src/styles.css:157]: This is so much clearer!!!

File: test/RecipeUtils-test.js [hunk 5]
```diff
@@ -6,20 +6,28 @@ import { recipesToCook, saveRecipe } from '../src/userUtils';
 
 
 describe('recipesFromTag', () => {
+  let save, save2, expected, names;
+
+  beforeEach(function() {
```
Comment (andreasorensen) [inline test/RecipeUtils-test.js:11]: YES!!!

**Review by andreasorensen — APPROVED** [review]

**Review by hvohr — APPROVED** [review]
this looks awesome, very clean and dry

**Pull request merged**